src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/graphs/AdaptiveChainPruner.java

```diff
@@ -29,41 +33,105 @@ public AdaptiveChainPruner(final double initialErrorProbability, final double lo
 
         final BaseGraph<V,E> graph = chains.get(0).getGraph();
 
+
+
         Collection<Path<V,E>> probableErrorChains = likelyErrorChains(chains, graph, initialErrorProbability);
         final int errorCount = probableErrorChains.stream().mapToInt(c -> c.getLastEdge().getMultiplicity()).sum();
         final int totalBases = chains.stream().mapToInt(c -> c.getEdges().stream().mapToInt(E::getMultiplicity).sum()).sum();
         final double errorRate = (double) errorCount / totalBases;
 
-        return likelyErrorChains(chains, graph, errorRate);
+        return likelyErrorChains(chains, graph, errorRate).stream().filter(c -> !c.getEdges().stream().anyMatch(BaseEdge::isRef)).collect(Collectors.toList());
```

Comment:
This is good, I *think* this resolves my previous concern about potentially pruning ref edges. I still think it would be prudent to adapt the test you have below to include a reference path that could/should be pruned otherwise just to be sure.

Reply:
Done by checking that ref path is assembled in `testAdaptivePruning` and by giving it's data provider a case where `altFraction` is 1.0